README.md

```diff
@@ -7,7 +7,41 @@ This repository provides KubeVirt-specific Tekton tasks, which focus on:
 - Executing commands in VMs
 - Manipulating disk images with virt-customize [WIP]
 
-## Usage and Deployment
+## Deployment
+
+### On Kubernetes
+
+In order to install the KubeVirt Tekton tasks in the active namespace you need to apply [this manifest](https://raw.githubusercontent.com/kubevirt/kubevirt-tekton-tasks/main/manifests/kubernetes/namespace-rbac/kubevirt-tekton-tasks.yaml).
+You have to repeat this for every namespace in which you'd like to run the tasks.
+
+```bash
+kubectl apply -f https://raw.githubusercontent.com/kubevirt/kubevirt-tekton-tasks/main/manifests/kubernetes/namespace-rbac/kubevirt-tekton-tasks.yaml
+```
+
+If the pipeline needs to create/access resources (VMs, PVCs, etc.) in a different namespace other than the one the pipeline runs in, the following commands need to be used instead.
+
+```bash
+TARGET_NAMESPACE="$(kubectl config view --minify --output 'jsonpath={..namespace}')"
```

Comment:
- let `TARGET_NAMESPACE="X"` and install the tasks
```
wget -qO - https://raw.githubusercontent.com/kubevirt/kubevirt-tekton-tasks/main/manifests/kubernetes/cluster-rbac/kubevirt-tekton-tasks.yaml | sed "s/TARGET_NAMESPACE/$TARGET_NAMESPACE/" | kubectl apply -f -
```

- create a pipeline/run in namespace `X` with following steps
  - use `create-datavolume-from-manifest` task to create DataVolume in namespace `Z` (specified in DV metadata) with PVC source from namespace `Y`
  - use `create-vm-from-manifest` task  to create a VM in namespace `Z` with a DV from previous step attached

Reply:
What should I do if my pipeline needs to create DVs and VMs in both Y and Z?

Reply:
That would depend on where the pipeline/run would be created. If in X, it would be very similar.

- let `TARGET_NAMESPACE="X"` and install the tasks
```
wget -qO - https://raw.githubusercontent.com/kubevirt/kubevirt-tekton-tasks/main/manifests/kubernetes/cluster-rbac/kubevirt-tekton-tasks.yaml | sed "s/TARGET_NAMESPACE/$TARGET_NAMESPACE/" | kubectl apply -f -
```

- create a pipeline/run in namespace `X` with following steps
   - use `create-datavolume-from-manifest` and `create-vm-from-manifest` tasks and insert VM / DV manifests with Y/Z namespaces accordingly. Y/Z are specified either through the manifest metadata or through the task namespace parameter.

Reply:
Do I understand correctly: once the admin installed the cluster-wide tasks to target namespace `X`, whoever can create a task on `X` can create a dv on any namespace? As a sysadmin, I'm not sure I'd like this security model

Reply:
This is a good point. Yes, the responsibility is on the sysadmin here - in which namespaces he will deploy these cluster privileges.

Maybe I should add a warning/explanation there that it will elevate the privileges and should be used for admin accounts only?

I would expect the sysadmin to tailor the correct privileges to their specific usecase - so if a user needs to access just two namespaces the admin will create the right RoleBindings just for these two namespaces.

Reply:
edited

Reply:
I don't know too much about your suggestions; as an admin, I'd like to know that users cannot abuse the pipeline to do things they typically are not allowed to. I would like to understand how a user with access to namespace X and Y can create a pipeline in X modifying object in Y, but not in Z.

Reply:
This arcticle gives a good explanation on how it could work: https://medium.com/faun/kubernetes-rbac-use-one-role-in-multiple-namespaces-d1d08bb08286 . I have not tried it yet, but don't see a big issue with it, except that the admin have to tailor the YAML to have the right RoleBindings for right namespaces.

I see two options from here on what to do:

1. We remove the cluster-rbac option from the README/releases and just tell the admins that they have to customize it if they want more. Which is quite painful for admins who just want to automate multiple namespaces in admin accounts.
2. Write a nice warning about it as I mentioned.

Reply:
ad 1. Maybe this one could be a good option if I write a good sub-README about the RBAC options. I quite like that it would cut down the number of release YAMLs from 4 to 2.

Thoughts?

Reply:
edited

Reply:
I don't know enough about k8s RBAC myself; I'm looking up to you to present a reasonable story for users. I want to have a pipeline that modifies several namespaces, but not others. What should I do? I'm willing to define a service account and set its RBAC if you tell me how.

Reply:
Ok, I will take that into account and try to explore the option 1.

Reply:
I am exploring this approach in https://github.com/kubevirt/kubevirt-tekton-tasks/pull/64 and  I can link the RBAC doc to the main README once it gets merged.